src/templates.hpp

```diff
@@ -176,14 +176,19 @@ void generic_sph(
 
     const auto size_y = (l_max + 1) * (l_max + 1);
     const auto size_q = (l_max + 1) * (l_max + 2) / 2;
-    #pragma omp parallel
+
+    // gets an index to some factors that enter the Qlm iteration, 
+    // and are pre-computed together with the prefactors
+    const double *qlmfactor = prefactors+size_q;
+
+    #pragma omp parallel 
     {
         // thread-local storage arrays for Qlm (modified associated Legendre
         // polynomials) and terms corresponding to (scaled) cosine and sine of
         // the azimuth
-        double *q = (double *)malloc(sizeof(double) * (l_max + 1) * (l_max + 2) / 2);
-        double *c = (double *)malloc(sizeof(double) * (l_max + 1));
-        double *s = (double *)malloc(sizeof(double) * (l_max + 1));
+        double *q = new double[(l_max + 1) * (l_max + 2) / 2];
```

Comment:
This could also be a `std::vector<double>`, removing  the need to remember to free/delete at the end.

Reply:
I didn't know if it made anything less compatible. If not, happy to go with it!

Reply:
I think if we assume C++17, we can use `std::vector` 😄 I'll do the change